### PR TITLE
[PF-700] simplify /status

### PR DESCRIPTION
**Pull request by ddietterich**
This simplifies the /status endpoint and the implementation. The high points:
- /status endpoint returns no content; it either returns 204 or 500 depending on the state
- Some status-check variables were constants and one was in a `@Value`. Consolidated all of them into StatusCheckConfiguration and set that.
- Added an `enabled` configuration, so the status check would not be scheduled in unit tests.
- Removed non-critical status checks from RBS and TDR

Drive-by: I renamed the test databases from the generic `testdb`, `wm` and `stairwaylib` to `wsm_db` and `wsm_stairway`. That names them in the same way as janitor does. If you use a persistent local Postgres (like I do), you can drop the old databases and rerun the script `service/local-dev/local-postgres-init.sql`. I did this because I have a lot of databases in my local Postgres and want to know which one goes with which service. I can pull this into a separate PR, based on review.

### CODE REVIEW

File: service/src/test/java/bio/terra/workspace/service/status/WorkspaceManagerStatusServiceTest.java [hunk 1]
```diff
@@ -2,72 +2,54 @@
...
-    doReturn(new ApiSystemStatusSystems().ok(false).addMessagesItem("Buffer down"))
-        .when(mockBufferService)
-        .status();
-    // Manually check subsystems, since @Scheduled doesn't work nicely in unit tests.
-    statusService.checkSubsystems();
-    assertFalse(statusService.getCurrentStatus().isOk());
-    Map<String, ApiSystemStatusSystems> subsystemStatus =
-        statusService.getCurrentStatus().getSystems();
-    assertFalse(subsystemStatus.get("Buffer").isOk());
+  void testCriticalFailureNotOk() {
```
Comment (zloery) [inline service/src/test/java/bio/terra/workspace/service/status/WorkspaceManagerStatusServiceTest.java:50]: nit: rename this, we no longer have "critical" vs. "non-critical" failures

**Review by zloery — APPROVED** [review]
LGTM, probably worth adding a bullet point to the PR description that this also removes non-critical status checks (RBS, TDR).
thanks for the logging doc update!

**Review by jaycarlton — APPROVED** [review]

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -201,7 +201,43 @@ Incompatible changes require incrementing the major version number. In our curre
...
+You can do this in any of the many ways to provide properties to Spring. A simple way is
+by setting the environment variable as:
+```sh
+export spring_profiles_include=human-readable-logging
+```
+
+### Controlling Log Level
+By default logging at INFO level and above are displayed. You can change the logging level
+by setting yet another Spring property. For example, you can set all things terra to log
+at debug level by adding this to a property YAML file:
```
[the remark this answers is not in the document]
Comment (ddietterich) [reply DEVELOPMENT.md:233]: Good catch. I added some detail.

File: service/local-dev/sql_validate.sh [hunk 3]
```diff
@@ -3,4 +3,4 @@
 # validate postgres
 echo "sleeping for 5 seconds during postgres boot..."
 sleep 5
-PGPASSWORD=wmpwd psql --username wmuser -d wm -c "SELECT VERSION();SELECT NOW()"
+PGPASSWORD=dbpwd psql --username dbuser -d wsm_db -c "SELECT VERSION();SELECT NOW()"
```
[the remark this answers is not in the document]
Comment (ddietterich) [reply service/local-dev/sql_validate.sh:6]: There was disagreement across different files about the user/password. There was no right answer.

File: service/local-dev/values.yaml.template [hunk 4]
```diff
@@ -2,3 +2,4 @@ imageConfig:
   imagePullPolicy: IfNotPresent
 global:
   terraEnv: ENV
+initDB: true
```
[the remark this answers is not in the document]
Comment (ddietterich) [reply service/local-dev/values.yaml.template:5]: Initializes the databases on deployment.

File: service/src/main/java/bio/terra/workspace/app/configuration/external/StatusCheckConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  /** Rate at which the status check is performed and the results are cached. */
+  private int pollingIntervalSeconds;
+
+  /** Seconds to wait after startup to begin status check polling */
+  private int startupWaitSeconds;
+
+  /**
+   * Allowed staleness of cached status check information before the service is marked as not ready.
+   * In seconds. For example, if polling interval is 30 seconds, staleness might be set to 65
+   * seconds to allow for one interval to be missed before declaring not ready.
```
[the remark this answers is not in the document]
Comment (ddietterich) [reply service/src/main/java/bio/terra/workspace/app/configuration/external/StatusCheckConfiguration.java:23]: Yes. I expanded the comment.

File: service/src/main/java/bio/terra/workspace/service/status/BaseStatusService.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  private final ConcurrentHashMap<String, Supplier<Boolean>> statusCheckMap;
+  /** scheduler */
+  private final ScheduledExecutorService scheduler;
+
+  public BaseStatusService(StatusCheckConfiguration configuration) {
+    this.configuration = configuration;
+    this.statusCheckMap = new ConcurrentHashMap<>();
+
+    this.statusOk = new AtomicBoolean(false);
+    this.lastStatusUpdate = Instant.now();
```
[the remark this answers is not in the document]
Comment (ddietterich) [reply service/src/main/java/bio/terra/workspace/service/status/BaseStatusService.java:46]: Uh oh. That is a bug. Not good that it didn't show up in the tests.

**Pull request merged**